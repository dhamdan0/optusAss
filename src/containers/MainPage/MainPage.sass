@import "../../style"

.mainPage
  margin-block-start: 5rem
  text-align: center
  max-width: 700px
  margin-inline: auto

  &__progressBars
    margin-block-start: 4rem

  &__progressBarControl
    display: flex
    align-items: center
    margin-inline: 1rem
    margin-block: 2rem
    flex-wrap: wrap

    @media (max-width: $tablet)
      justify-content: center

