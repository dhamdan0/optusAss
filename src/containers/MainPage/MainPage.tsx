import './MainPage.sass'
import ProgressBar from '../../components/ProgressBar/ProgressBar'
import ButtonsControl from '../../components/ButtonsControl/ButtonsControl'
import React, { type ChangeEvent, useCallback, useState } from 'react'
import SelectBox from '../../components/SelectBox/SelectBox'

import data from '../../data/data.json'

const cssPrefix: string = 'mainPage'

const selectOptions = data.progressBarOptions.map((it) => ({ value: it.key, label: it.label }))

function MainPage () {
  const [progressBars, setProgressBars] = useState(data.progressBarOptions)
  const [selectedProgressbar, setSelectedProgressbar] = useState(1)

  const handleChangeProgressBar = useCallback((e: ChangeEvent<HTMLSelectElement>) => {
    setSelectedProgressbar(+e.target.value)
  }, [setSelectedProgressbar])

  const handleButtonClick = useCallback((val: number) => {
    const list = [...progressBars]
    const progressbar = list.find(it => it.key === selectedProgressbar)
    if (progressbar != null) {
      progressbar.value = Math.max(+progressbar.value + val, 0)
    }
    setProgressBars(list)
  }, [selectedProgressbar, setProgressBars, progressBars])

  return (
      <div className={cssPrefix}>
          <h1>Progress Bars</h1>

        <div className={`${cssPrefix}__progressBars`}>
          {progressBars.map((it) => {
            return <ProgressBar key={it.key} progress={it.value}/>
          })}
        </div>

        <div className={`${cssPrefix}__progressBarControl`}>
          <SelectBox
              options={selectOptions}
              selectedValue={selectedProgressbar}
              onChangeOption={handleChangeProgressBar}
          />
          <ButtonsControl
              values={data.buttonsOptions}
              onButtonClick={handleButtonClick}
          />
        </div >
      </div>
  )
}

export default MainPage
