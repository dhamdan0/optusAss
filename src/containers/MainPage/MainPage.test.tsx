import React from 'react'
import { render, screen } from '@testing-library/react'
import MainPage from './MainPage'

jest.mock('../../components/SelectBox/SelectBox', () => 'SelectBox')
jest.mock('../../components/ButtonsControl/ButtonsControl', () => 'ButtonsControl')
jest.mock('../../components/ProgressBar/ProgressBar', () => 'ProgressBar')

function testComponentWithProps () {
  const container = render(<MainPage />)
  expect(container).toMatchSnapshot()
}

describe('MainPage', () => {
  it('renders page title', () => {
    render(<MainPage />)
    const header = screen.getByText(/Progress Bars/i)
    expect(header).toBeInTheDocument()
  })

  it('renders the component', () => {
    testComponentWithProps()
  })
})
