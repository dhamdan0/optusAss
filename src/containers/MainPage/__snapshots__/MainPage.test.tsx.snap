// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mainPage"
      >
        <h1>
          Progress Bars
        </h1>
        <div
          class="mainPage__progressBars"
        >
          <progressbar
            progress="25"
          />
          <progressbar
            progress="50"
          />
          <progressbar
            progress="75"
          />
        </div>
        <div
          class="mainPage__progressBarControl"
        >
          <selectbox
            options="[object Object],[object Object],[object Object]"
            selectedvalue="1"
          />
          <buttonscontrol
            values="-25,-10,10,25"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mainPage"
    >
      <h1>
        Progress Bars
      </h1>
      <div
        class="mainPage__progressBars"
      >
        <progressbar
          progress="25"
        />
        <progressbar
          progress="50"
        />
        <progressbar
          progress="75"
        />
      </div>
      <div
        class="mainPage__progressBarControl"
      >
        <selectbox
          options="[object Object],[object Object],[object Object]"
          selectedvalue="1"
        />
        <buttonscontrol
          values="-25,-10,10,25"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
